server/src/main/java/io/unitycatalog/server/persist/ModelRepository.java

```diff
@@ -345,17 +358,426 @@ public void deleteRegisteredModel(String fullName, boolean force) {
+    LOGGER.info("Deleting model version: " + fullName + "/" + version);
+    String[] parts = fullName.split("\\.");
+    if (parts.length != 3) {
+      throw new BaseException(
+          ErrorCode.INVALID_ARGUMENT, "Invalid registered model name: " + fullName);
+    }
+    String catalogName = parts[0];
+    String schemaName = parts[1];
+    String registeredModelName = parts[2];
+    RegisteredModelInfo rmInfo = getRegisteredModel(fullName);
+    if (rmInfo == null) {
+      throw new BaseException(
+          ErrorCode.INVALID_ARGUMENT, "Invalid registered model name: " + fullName);
+    }
+    try (Session session = SESSION_FACTORY.openSession()) {
+      Transaction tx = session.beginTransaction();
+      try {
+        UUID schemaId = getSchemaId(session, catalogName, schemaName);
+        RegisteredModelInfoDAO existingRegisteredModel =
+            getRegisteredModelDao(session, schemaId, registeredModelName);
+        if (existingRegisteredModel == null) {
+          throw new BaseException(
+              ErrorCode.NOT_FOUND, "Registered model not found: " + registeredModelName);
+        }
+        deleteModelVersion(session, existingRegisteredModel.getId(), fullName, version);
+        tx.commit();
+      } catch (RuntimeException e) {
+        if (tx != null && tx.getStatus().canRollback()) {
+          tx.rollback();
+        }
+        throw e;
+      }
+    }
+  }
+
+  public void deleteModelVersion(
+      Session session, UUID registeredModelId, String fullName, long version) {
+    ModelVersionInfoDAO modelVersionInfoDAO =
+        getModelVersionDao(session, registeredModelId, version);
+    if (modelVersionInfoDAO == null) {
+      throw new BaseException(
+          ErrorCode.NOT_FOUND, "Model version not found: " + fullName + "/" + version);
+    }
+    session.remove(modelVersionInfoDAO);
+  }
+
+  public ModelVersionInfo finalizeModelVersion(FinalizeModelVersion finalizeModelVersion) {
+    if (finalizeModelVersion.getFullName() == null) {
+      throw new BaseException(ErrorCode.INVALID_ARGUMENT, "No model specified.");
+    }
+    if (finalizeModelVersion.getVersion() == null) {
+      throw new BaseException(ErrorCode.INVALID_ARGUMENT, "No version specified.");
+    }
+
+    String fullName = finalizeModelVersion.getFullName();
+    Long version = finalizeModelVersion.getVersion();
+    LOGGER.info("Finalize Model Version: " + fullName + "/" + version);
+    ModelVersionInfo modelVersionInfo;
+
+    Transaction tx;
+    try (Session session = SESSION_FACTORY.openSession()) {
+      String[] parts = RepositoryUtils.parseFullName(fullName);
+      String catalogName = parts[0];
+      String schemaName = parts[1];
+      String registeredModelName = parts[2];
+      tx = session.beginTransaction();
+      try {
+        // Get the registered model record from the database
+        UUID schemaId = getSchemaId(session, catalogName, schemaName);
+        RegisteredModelInfoDAO existingRegisteredModel =
```

Comment:
can we move getting the parent RegisteredModelDAO and subsequently ModelVersionDAO to a common function, it's repeated everywhere. The function can just return just the ModelVersionDAO.